src/color.rs

```diff
@@ -48,6 +78,11 @@ impl ColorType {
         }
     }
 
+    /// Returns the number of bytes contained in a channel per pixel of `ColorType` ```c```
+    pub fn bytes_per_channel(self) -> u8 {
```

Comment:
This will essentially lock the `ColorType` to use the same amount of bytes for all channels, which will probably holds true for a good while. Still, I wonder if it wouldn't be simpler to have a local method in `png` that matches the color types themselves or does this arithemtic instead.

Reply:
I thought it better put this arithmetic in a central location so if we are able to support inhomogeneous channel depths later we could pick most of them up by looking for usages. But I agree looking at it now it might have the opposite effect of making homogeneity look like a safe assumption. I'll move the logic to `png`.